obspy/taup/taup_geo.py

```diff
@@ -0,0 +1,157 @@
+    :param station_latitude_in_deg: Station location latitude in degrees
+    :type station_latitude_in_deg: float
+    :param station_longitude_in_deg: Station location longitude in degrees
+    :type station_longitude_in_deg: float
+    :param radius_of_earth_in_km: Radius of the Earth in km
+    :type radius_of_earth_in_km: float
+    :param flattening_of_earth: Flattening of earth (0 for a sphere)
+    :type station_longitude_in_deg: float
+
+    :return: List of ``Arrival`` objects, each of which has the time,
+        corresponding phase name, ray parameter, takeoff angle, etc. as
+        attributes.
+    :rtype: :class:`Arrivals`
+    """
+    try:
+        import geographiclib.geodesic as geod
+        ellipsoid = geod.Geodesic(a=radius_of_earth_in_km * 1000.0,
+                                  f=flattening_of_earth)
+        g = ellipsoid.Inverse(source_latitude_in_deg, source_longitude_in_deg,
+                              station_latitude_in_deg,
+                              station_longitude_in_deg)
+        azimuth = g['azi1']
+        line = ellipsoid.Line(source_latitude_in_deg,
+                              source_longitude_in_deg, azimuth)
+
+        # We may need to update many arrival objects
+        # and each could have pierce points and a
+        # path
+        for arrival in arrivals:
+
+            if arrival.pierce is not None:
+                pathList = []
+                for pierce_point in arrival.pierce:
+                    pos = line.ArcPosition(np.degrees(pierce_point['dist']))
+                    diffTDG = np.array([(
+                        pierce_point['p'],
+                        pierce_point['time'],
+                        pierce_point['dist'],
+                        pierce_point['depth'],
+                        pos['lat2'],
+                        pos['lon2'])], dtype=TimeDistGeo)
+                    pathList.append(diffTDG)
+                arrival.pierce = np.concatenate(pathList)
+
+            if arrival.path is not None:
+                pathList = []
+                for path_point in arrival.path:
+                    pos = line.ArcPosition(np.degrees(path_point['dist']))
+                    diffTDG = np.array([(
+                        path_point['p'],
+                        path_point['time'],
+                        path_point['dist'],
+                        path_point['depth'],
+                        pos['lat2'],
+                        pos['lon2'])], dtype=TimeDistGeo)
+                    pathList.append(diffTDG)
+                arrival.path = np.concatenate(pathList)
+
+    except ImportError:
+        # geographiclib is not installed ...
+        # and  obspy/geodetics does not help much
+        msg = "Not able to evaluate positions on path or pierce points. " + \
+              "Arrivals object will not be modified. " + \
+              "Install the Python module 'geographiclib' to solve this issue."
+        warnings.warn(msg)
```

Comment:
I wouldn't warn here but raise an ImportError instead, with the message you have there.